api/src/game_state_server.py

```diff
@@ -189,17 +189,21 @@ async def process_updates(
                                 'error', f'Received a bad token: {data}', request_id
                             ),
                         )
-                    if not self._is_valid_position(token, room_id):
-                        logger.info(
-                            f'Token {token.id} cannot move to occupied position'
-                        )
-                        yield Message(
-                            {client_id},
-                            MessageContents(
-                                'error', 'That position is occupied, bucko', request_id
-                            ),
-                        )
-                    self._create_or_update_token(token, room_id)
+                    else:
```

Comment:
I think the organization of this chain can be improved even more. 

it boils down to:
```
if bad condition:
    yield error
else:
   if good condition:
     do thing
   else:
     yield error
```

I think it would be more readable if it went
```
if bad condition:
  yield error
elif other bad condition:
  yield error
else
  do thing
```
That way it still reads like the early exits you were doing before switching to yields

Reply:
Sure.